apps/server/src/modules/tool/tool.module.ts

```diff
@@ -1,18 +1,40 @@
 import { Module } from '@nestjs/common';
 import { LtiToolRepo, PseudonymsRepo } from '@shared/repo';
-import { Lti11ResponseMapper } from '@src/modules/tool/mapper/lti11-response.mapper';
 import { LoggerModule } from '@src/core/logger';
-import { LtiToolService } from '@src/modules/tool/service/lti-tool.service';
 import { SchoolModule } from '@src/modules/school';
-import { ToolController } from '@src/modules/tool/controller/tool.controller';
+import { ExternalToolRepoMapper } from '@shared/repo/externaltool/external-tool.repo.mapper';
+import { ExternalToolRepo } from '@shared/repo/externaltool/external-tool.repo';
+import { AuthorizationModule } from '@src/modules/authorization';
+import { OauthProviderServiceModule } from '@shared/infra/oauth-provider';
+import { EncryptionModule } from '@shared/infra/encryption';
 import { Lti11Uc } from './uc/lti11.uc';
 import { LtiRoleMapper } from './mapper/lti-role.mapper';
 import { Lti11Service } from './service/lti11.service';
 import { UserModule } from '../user';
+import { ToolController } from './controller/tool.controller';
+import { Lti11ResponseMapper } from './mapper/lti11-response.mapper';
+import { ExternalToolUc } from './uc/external-tool.uc';
+import { ExternalToolService } from './service/external-tool.service';
+import { ExternalToolRequestMapper } from './mapper/external-tool-request.mapper';
+import { ExternalToolResponseMapper } from './mapper/external-tool-response.mapper';
 
 @Module({
-	imports: [UserModule, LoggerModule, SchoolModule],
+	imports: [UserModule, LoggerModule, SchoolModule, AuthorizationModule, OauthProviderServiceModule, EncryptionModule],
 	controllers: [ToolController],
-	providers: [Lti11Service, Lti11Uc, LtiRoleMapper, Lti11ResponseMapper, LtiToolRepo, PseudonymsRepo, LtiToolService],
+	providers: [
```

Comment:
is there a ticket to split this into api and internal module?

Reply:
Not atm, is there an official statement to do that? Our current info is this:

[SevenWaysDP](https://github.com/SevenWaysDP) [3 days ago](https://github.com/hpi-schul-cloud/schulcloud-server/pull/3710#discussion_r1026103607)
Please split it to tool-api.module.ts and tool.module.ts
For example look Lesson Module

Member
@[alweber-cap](https://github.com/alweber-cap) alweber-cap [3 days ago](https://github.com/hpi-schul-cloud/schulcloud-server/pull/3710#discussion_r1026428254)
Is this already set as the target-architecture? Afaik we did not yet decide on the final structure.

If this is the final structure, then i would like to have do the separation in a follow-up ticket, since we will have further tasks here. Separating them here will increase the complexity of this even more.

Reply:
yes that one was decided, and has been documented here https://hpi-schul-cloud.github.io/schulcloud-server/additional-documentation/nestjs-application/software-architecture.html#modules

Reply:
whats still open is the horizontal architecture (what modules we want to have exactly), and how exactly domain objects are to be handled within the modules (though we already know for sure that the old entities should never leave the data layer, and definetly never leave the module)

Reply:
i might have missed the final decision on that change. we will create a follow-up ticket to split it up

Reply:
Ticket ready: https://ticketsystem.dbildungscloud.de/browse/N21-465